packages/web3-eth-contract/src/contract.ts

```diff
@@ -139,6 +143,73 @@ export class Contract<Abi extends ContractAbi>
 		});
 	}
 
+	public deploy(deployOptions?: {
+		data?: HexString;
+		arguments?: ContractConstructor<Abi>['Inputs'];
+	}) {
+		const abi = this._jsonInterface.find(j => j.type === 'constructor');
+
+		if (!abi) {
+			throw new Web3ContractError('No constructor interface found.');
+		}
+
+		if (!deployOptions?.data) {
+			throw new Web3ContractError('No data provided.');
+		}
+		const data = deployOptions?.data ?? this.options.data;
+		const args = deployOptions?.arguments ?? [];
+		validator.validate(['string'], args);
+
+		const contractOptions = { ...this.options, data };
+
+		return {
+			arguments: args,
+			// TODO: Use `web3-eth-tx` package to return `PromiEvent` instead.
+			send: async (options?: PayableCallOptions) => {
+				const modifiedOptions = { ...options };
+				delete modifiedOptions.to;
+
+				const txHash = await this._contractMethodSend(
+					abi as AbiFunctionFragment,
+					args,
+					modifiedOptions,
+					contractOptions,
+				);
+
+				return txHash;
+
+				// TODO: Use eth-tx functions to
```

Comment:
For tracking we should create issue for using Eth and Eth-Tx in contracts package.

Reply:
https://github.com/ChainSafe/web3.js/issues/4754